File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<Integer>();
        int n = temperatures.length;
        int[] ans  = new int [n];
        for(int i = 0; i < n; i++) {
            if(stack.isEmpty()) {
                stack.push(i);
            } else {
                while(stack.isEmpty() == false && temperatures[stack.peek()] < temperatures[i]) {
                    //System.out.println(temperatures[stack.peek()] + " is warmed by " + temperatures[i]);
                    ans[stack.peek()] = i - stack.peek();
                    stack.pop();
                }
                stack.push(i);
            }
        }
        return ans;
        
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    
    const n = temperatures.length;
    let increasingStack = [];
    let ans = new Array(n);
    ans.fill(0);
    for(let i = 0; i < n; i++) {
        if(increasingStack.length == 0) {
            increasingStack.push(i);
        } else {
            let top = increasingStack[increasingStack.length - 1];
            while(increasingStack.length && temperatures[i] > temperatures[top]) {
                ans[top] = i - top;
                increasingStack.pop(i);
                top = increasingStack[increasingStack.length - 1];
            }
            increasingStack.push(i)
        }
    }
    return ans;
    
    
};